pkg/controller/machinedeployment/machinedeployment_controller.go

```diff
@@ -123,6 +123,13 @@ func (r *ReconcileMachineDeployment) Reconcile(request reconcile.Request) (recon
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+
+	// Ignore deleted MachineDeployments, this can happen when foregroundDeletion
+	// is enabled
+	if d.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
```

Comment:
Yeah, didn't see that because its further down the line. Similiarily to the MS, do you know why some code is still executed for a deleted MD; like the validation, defaulting, possibly getting the cluster, getting machinesets and machines?